README.md

```diff
@@ -75,6 +75,73 @@ The PMEM-CSI driver supports running in different modes, which can be controlled
 
 * **_Unified_** mode is intended to run the driver in a single host, mostly for testing the driver in a non-clustered environment.
 
+### Volume Persistency
+
+Normally, volumes are provided by a storage backend that is independent
+of a particular node. When a node goes offline, the volume can be
+mounted elsewhere. With PMEM-CSI, volumes are *local* and thus can
+only be used on the node where they were created.
+
+This limitation needs to be considered when designing and deploying
+applications that are to use local storage:
+
+* Persistent Volumes  
+A volume gets created independently of the application, on some node
+where there is enough free space. Applications using such a volume are
+then forced to run on that node and cannot run when the node is
+down. Data is retained until the volume gets deleted.
+* Ephemeral Volumes  
+An application is assigned to a node and when it starts to run, a new
+volume is created for it on that node. When the application stops, the
+volume is deleted.
+* Cache Volumes  
+Volumes are created on a certain set of nodes, each with its own local
+data. Applications are started on those nodes and then get to use the
+volume on their node. Data persists across application restarts. This
+is useful when the data is only cached information that can be
+discarded and reconstructed at any time *and* the application can
+reuse existing local data when restarting.
+* Delayed Persistent Volumes  
```

Comment:
We might deprecate or merge this with *Persistent Volumes*, In implementation side, when there is enough information(csi.TopologyRequirement) we respect that and provision the volume accordingly, else we pick random node to provision.

Reply:
What is the correct way in your opinion. Choosing node before hand without knowing Pod's other constraints might lead to more problems.

Reply:
IMHO choosing the location of data is the most important choice, because it is final and cannot be changed afterwards. I think someone who sets up a cluster where some apps use persistent local data must have a plan for where those apps are meant to run. Communicating that plan to the pmem-csi driver via pod scheduling would work technically, but it seems rather indirect and non-obvious. I would prefer to have some explicit annotation on the PVC itself.

But I might be wrong - we would have to ask users.